### PR TITLE
hwdef: Add a new hwdef param for AEROFOX H7  flight control

### PR DESCRIPTION
**Pull request by AEROFOXtech**
1. Add a new flight control board hwdef for AEROFOX_H7
2. Create a new board ID for AEROFOX_H7 flight control board
3. Add AEROFOX_H7 bootloader to Tools

### CODE REVIEW

File: Tools/AP_Bootloader/board_types.txt
Comment (andyp1per) [inline Tools/AP_Bootloader/board_types.txt:408]: This needs to be in a separate commit

Comment (andyp1per) [reply Tools/AP_Bootloader/board_types.txt:408]: Ditto the bootloader

File: libraries/AP_HAL_ChibiOS/hwdef/AEROFOX-H7/defaults.parm
Comment (andyp1per) [inline libraries/AP_HAL_ChibiOS/hwdef/AEROFOX-H7/defaults.parm:7]: You should not set parameters that are already the default as it consumes flash space. Ideally put these in the hwdef

File: libraries/AP_HAL_ChibiOS/hwdef/AEROFOX-H7/hwdef-bl.dat
Comment (andyp1per) [inline libraries/AP_HAL_ChibiOS/hwdef/AEROFOX-H7/hwdef-bl.dat:8]: Use AP_HW_AEROFOX_H7

File: libraries/AP_HAL_ChibiOS/hwdef/AEROFOX-H7/hwdef.dat
Comment (andyp1per) [inline libraries/AP_HAL_ChibiOS/hwdef/AEROFOX-H7/hwdef.dat:8]: Use AP_HW_AEROFOX_H7

File: libraries/AP_HAL_ChibiOS/hwdef/AEROFOX-H7/hwdef.dat
Comment (andyp1per) [inline libraries/AP_HAL_ChibiOS/hwdef/AEROFOX-H7/hwdef.dat:96]: Make these 16Mhz if you can

File: libraries/AP_HAL_ChibiOS/hwdef/AEROFOX-H7/hwdef.dat
Comment (andyp1per) [inline libraries/AP_HAL_ChibiOS/hwdef/AEROFOX-H7/hwdef.dat:100]: Ideally you would use SDMMC here

File: libraries/AP_HAL_ChibiOS/hwdef/AEROFOX-H7/hwdef.dat
Comment (andyp1per) [inline libraries/AP_HAL_ChibiOS/hwdef/AEROFOX-H7/hwdef.dat:124]: Should make these bdshot, should be easy

**Review by andyp1per — CHANGES_REQUESTED** [review]
Some changes needed

File: Tools/AP_Bootloader/board_types.txt
Comment (laoshen14789) [reply Tools/AP_Bootloader/board_types.txt:408]: I'll submit to modify it in the next commit

File: libraries/AP_HAL_ChibiOS/hwdef/AEROFOX-H7/defaults.parm
Comment (laoshen14789) [reply libraries/AP_HAL_ChibiOS/hwdef/AEROFOX-H7/defaults.parm:7]: Thank you for your advice

File: libraries/AP_HAL_ChibiOS/hwdef/AEROFOX-H7/hwdef.dat
Comment (laoshen14789) [reply libraries/AP_HAL_ChibiOS/hwdef/AEROFOX-H7/hwdef.dat:100]: Thank you for your advice, but we don't have enough GPIO ports on our hardware to use SDMMC

**Review by Hwurzburg — CHANGES_REQUESTED** [review]
needs readme ala https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_HAL_ChibiOS/hwdef/KakuteH7-Wing/README.md  with all info and images

**Comment by Hwurzburg** [comment]
@AEROFOXtech I really need each connector's pinout to review...also views of any solder pads/thru holes with their pin labels

File: libraries/AP_HAL_ChibiOS/hwdef/AEROFOX-H7/README.md
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/AEROFOX-H7/README.md:95]:
```suggestion
- BATT_AMP_PERVLT should be set as required by the specific monitor used

```

File: libraries/AP_HAL_ChibiOS/hwdef/AEROFOX-H7/README.md
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/AEROFOX-H7/README.md:101]:
```suggestion
is usually disabled and an external compass used mounted to minimize motor current effects.

```

File: libraries/AP_HAL_ChibiOS/hwdef/AEROFOX-H7/defaults.parm
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/AEROFOX-H7/defaults.parm:1]: remove, unless you are supplying a Dronecan airspeed sensor with the autopilot to purchasers.

File: libraries/AP_HAL_ChibiOS/hwdef/AEROFOX-H7/defaults.parm
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/AEROFOX-H7/defaults.parm:2]: all BATT/BATT2 params should be moved to the hwdef file

File: libraries/AP_HAL_ChibiOS/hwdef/AEROFOX-H7/defaults.parm
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/AEROFOX-H7/defaults.parm:6]: unless this is going to only be used in QuadPlanes as a turn key system, remove this line

File: libraries/AP_HAL_ChibiOS/hwdef/AEROFOX-H7/hwdef.dat
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/AEROFOX-H7/hwdef.dat:15]: UART1 is SERIAL5 and should be GPS

File: libraries/AP_HAL_ChibiOS/hwdef/AEROFOX-H7/hwdef.dat
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/AEROFOX-H7/hwdef.dat:18]:
```suggestion
define DEFAULT_SERIAL5_PROTOCOL SerialProtocol_GPS
```

File: libraries/AP_HAL_ChibiOS/hwdef/AEROFOX-H7/hwdef.dat
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/AEROFOX-H7/hwdef.dat:45]: remove, not needed

File: libraries/AP_HAL_ChibiOS/hwdef/AEROFOX-H7/hwdef.dat
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/AEROFOX-H7/hwdef.dat:51]: remove, not needed

File: libraries/AP_HAL_ChibiOS/hwdef/AEROFOX-H7/hwdef.dat
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/AEROFOX-H7/hwdef.dat:57]: remove, not needed

File: libraries/AP_HAL_ChibiOS/hwdef/AEROFOX-H7/hwdef.dat
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/AEROFOX-H7/hwdef.dat:63]: remove, most telemi s 64K, the default

File: libraries/AP_HAL_ChibiOS/hwdef/AEROFOX-H7/hwdef.dat
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/AEROFOX-H7/hwdef.dat:69]: remove, not needed for RCIN protocols

File: libraries/AP_HAL_ChibiOS/hwdef/AEROFOX-H7/hwdef.dat
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/AEROFOX-H7/hwdef.dat:72]: we prefer that the second OTG use MAVLINK2 (the default) to avoid user confusion...using MAVLink SLCAN s better access to CAN

File: libraries/AP_HAL_ChibiOS/hwdef/AEROFOX-H7/hwdef.dat
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/AEROFOX-H7/hwdef.dat:211]: remove....does not exist

**Review by Hwurzburg — CHANGES_REQUESTED** [review]
you should add SD card I/O to bootloader code to allow firmware update from SD card
also I have several other review comments requiring changes

**Comment by Hwurzburg** [comment]
@AEROFOXtech is this ready for final re-review?you can request in the reviewers box upper right

**Review by Hwurzburg — CHANGES_REQUESTED** [review]
a few items missed still

File: libraries/AP_HAL_ChibiOS/hwdef/AEROFOX-H7/hwdef.dat
Comment (AEROFOXtech) [reply libraries/AP_HAL_ChibiOS/hwdef/AEROFOX-H7/hwdef.dat]: Do you mean to cancel only the baud rate configuration?

File: libraries/AP_HAL_ChibiOS/hwdef/AEROFOX-H7/hwdef.dat
Comment (AEROFOXtech) [reply libraries/AP_HAL_ChibiOS/hwdef/AEROFOX-H7/hwdef.dat]: Do you mean to cancel only the baud rate configuration?

Comment (Hwurzburg) [reply libraries/AP_HAL_ChibiOS/hwdef/AEROFOX-H7/hwdef.dat]: yes....comments apply to the last line shown before the comment

File: libraries/AP_HAL_ChibiOS/hwdef/AEROFOX-H7/hwdef.dat
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/AEROFOX-H7/hwdef.dat:71]:
```suggestion
# default the 2nd interface to MAVLINK2
define HAL_OTG2_PROTOCOL SerialProtocol_MAVLINK2
```
you removed the SLCAN but did not force the MAVLINK2....recommended, but not a blocker

**Review by Hwurzburg — CHANGES_REQUESTED** [review]
up to you but we recommend OTG2 be MAVLink2 instead of the default of SLCAN since SLCAN is not as good as using GCS's MAVLink CAN and having SLCAN2 for second USB port can frustrate new users trying to connect a GCS (since it will not on that port)

Also, this PR needs to be two commits....one titled Tools: add AEROFOX H7 bootloaders, with the tools library additions, and on titled hwdef: Add AEROFOX H7

File: libraries/AP_HAL_ChibiOS/hwdef/AEROFOX-H7/hwdef.dat
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/AEROFOX-H7/hwdef.dat:236]: probably should not enable 2nd monitor by default...just remove line 236

**Review by Hwurzburg — CHANGES_REQUESTED** [review]
except for one nit...looks good....
needs @andyp1per approval also

if you have tested, once Andy approve and you disable 2nd batt mon by default, we can mark for devcall and get it merged.....

File: libraries/AP_HAL_ChibiOS/hwdef/AEROFOX-H7/hwdef.dat
Comment (AEROFOXtech) [reply libraries/AP_HAL_ChibiOS/hwdef/AEROFOX-H7/hwdef.dat:236]: OK, I'll make the changes right away.

**Review by Hwurzburg — APPROVED** [review]

**Review by andyp1per — APPROVED** [review]

File: Tools/AP_Bootloader/board_types.txt
Comment (peterbarker) [inline Tools/AP_Bootloader/board_types.txt:436]: Why did you not follow these instructions?

**Review by peterbarker — CHANGES_REQUESTED** [review]
Please fill in the gaps, don't expand the range used.

File: Tools/AP_Bootloader/board_types.txt
Comment (AEROFOXtech) [reply Tools/AP_Bootloader/board_types.txt:436]: Sorry, I will fix these soon

**Pull request merged**
